python/cudf/dataframe/string.py

```diff
@@ -127,9 +127,72 @@ def cat(self, others=None, sep=None, na_rep=None):
             (same type as caller) of str dtype is returned.
         """
         from cudf.dataframe import Series, Index
+
         if isinstance(others, (Series, Index)):
+            '''
+            If others is just another Series/Index,
+            great go ahead with concatenation
+            '''
             assert others.dtype == np.dtype('object')
             others = others.data
+        elif utils.is_list_like(others) and others:
+            '''
+            If others is a list-like object (in our case lists & tuples)
+            just another Series/Index, great go ahead with concatenation.
+            '''
+
+            '''
+            Picking first element and checking if it really adheres to
+            list like conditions, if not we don't have the
+            necessity to scan the rest of the list.
+            '''
+            first = others[0]
+
+            if (utils.is_list_like(first) or
+                isinstance(first, (Series, Index, pd.Series, pd.Index))) \
```

Comment:
`Series` and `Index` types can't have lists as of now so you shouldn't check for those here.

Reply:
Discussing offline regarding this.